src/librustdoc/html/render/print_item.rs

```diff
@@ -1578,21 +1578,23 @@ fn item_static(w: &mut impl fmt::Write, cx: &mut Context<'_>, it: &clean::Item,
     write!(w, "{}", document(cx, it, None, HeadingOffset::H2)).unwrap();
 }
 
-fn item_foreign_type(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item) {
-    wrap_item(w, |w| {
-        w.write_str("extern {\n");
-        render_attributes_in_code(w, it, cx.tcx());
+fn item_foreign_type(w: &mut impl fmt::Write, cx: &mut Context<'_>, it: &clean::Item) {
+    let mut buffer = Buffer::new();
+    wrap_item(&mut buffer, |buffer| {
```

Comment:
Here render attributes expects a buffer so we might have to create one here or change the implementation of render_attributes in code to take both a buffer and an impl write
```rust
render_attributes_in_code(w, it, cx.tcx());
     |         ------------------------- ^ expected `&mut Buffer`, found `&mut impl fmt::Write`
     |         |
     |         arguments to this function are incorrect
     |

```

How we could modify render_attributess_in_code

```rust
enum WriteWrapper<'a> {
    Buffer(&'a mut html::format::Buffer),
    Writer(&'a mut dyn fmt::Write),
}

impl fmt::Write for WriteWrapper<'_> {
    fn write_str(&mut self, s: &str) -> fmt::Result {
        match self {
            WriteWrapper::Buffer(buffer) => buffer.write_str(s),
            WriteWrapper::Writer(writer) => writer.write_str(s),
        }
    }
}

fn render_attributes_in_code(w: &mut WriteWrapper<'_>, it: &clean::Item, tcx: TyCtxt<'_>) {
    for a in it.attributes(tcx, false) {
        write!(w, "<div class=\"code-attribute\">{}</div>", a);
    }
}

Reply:
Or you can update `render_attributes_in_code` to make it take a `impl Write` instead of `Buffer`. ;)

```rust
fn render_attributes_in_code(w: &mut Buffer, it: &clean::Item, tcx: TyCtxt<'_>) {
    for a in it.attributes(tcx, false) {
        write!(w, "<div class=\"code-attribute\">{}</div>", a);
    }
}
```

Reply:
(Unless I missed something obvious, `Buffer` implements `Write` so it should be the only change required?)

Reply:
We can actually remove the use of `Buffer` here and only rely on `write!()`.

Recently, I made a change (to be merged soon) on `render_attributes_in_code()` that doesn't have `Buffer` nor `impl Write` in its input. Instead, it returns `impl Display`, which then can be used for `write!()`.

[See the change here](https://github.com/rust-lang/rust/pull/111430/files#diff-40a0eb025da61717b3b765ceb7fab21d91af3012360e90b9f46e15a4047946faR1043-R1053)